docs/design/ovs-pipeline.md

```diff
@@ -1,1197 +1,1941 @@
+packets based on their destination IP addresses, as well as their reg marks or ct marks.
+
+If you dump the flows of this table, you may see the following:
 
 ```text
-table=70, priority=200,ip,reg0=0/0x80000,nw_dst=10.10.1.0/24 actions=goto_table:80
-table=70, priority=200,ip,reg0=0x2/0xffff,nw_dst=192.168.1.1 actions=goto_table:80
+1. table=L3Forwarding, priority=210,ip,nw_dst=10.10.0.1 actions=set_field:ba:5e:d1:55:aa:c0->eth_dst,set_field:0x20/0xf0->reg0,goto_table:L3DecTTL
+2. table=L3Forwarding, priority=210,ct_state=+rpl+trk,ct_mark=0x2/0xf,ip actions=set_field:ba:5e:d1:55:aa:c0->eth_dst,set_field:0x20/0xf0->reg0,goto_table:L3DecTTL
+3. table=L3Forwarding, priority=200,ip,reg0=0/0x200,nw_dst=10.10.0.0/24 actions=goto_table:L2ForwardingCalc
+4. table=L3Forwarding, priority=200,ip,nw_dst=10.10.1.0/24 actions=set_field:ba:5e:d1:55:aa:c0->eth_src,set_field:aa:bb:cc:dd:ee:ff->eth_dst,set_field:192.168.77.103->tun_dst,set_field:0x10/0xf0->reg0,goto_table:L3DecTTL
+5. table=L3Forwarding, priority=200,ip,reg0=0x200/0x200,nw_dst=10.10.0.24 actions=set_field:ba:5e:d1:55:aa:c0->eth_src,set_field:fa:b7:53:74:21:a6->eth_dst,goto_table:L3DecTTL
+6. table=L3Forwarding, priority=200,ip,reg0=0x200/0x200,nw_dst=10.10.0.25 actions=set_field:ba:5e:d1:55:aa:c0->eth_src,set_field:36:48:21:a2:9d:b4->eth_dst,goto_table:L3DecTTL
+7. table=L3Forwarding, priority=200,ip,reg0=0x200/0x200,nw_dst=10.10.0.26 actions=set_field:ba:5e:d1:55:aa:c0->eth_src,set_field:5e:b5:e3:a6:90:b7->eth_dst,goto_table:L3DecTTL
+8. table=L3Forwarding, priority=190,ct_state=-rpl+trk,ip,reg0=0x3/0xf,reg4=0/0x100000 actions=goto_table:EgressMark
+9. table=L3Forwarding, priority=190,ct_state=-rpl+trk,ip,reg0=0x1/0xf actions=set_field:ba:5e:d1:55:aa:c0->eth_dst,goto_table:EgressMark
+10. table=L3Forwarding, priority=190,ct_mark=0x10/0x10,reg0=0x202/0x20f actions=set_field:ba:5e:d1:55:aa:c0->eth_dst,set_field:0x20/0xf0->reg0,goto_table:L3DecTTL
+11. table=L3Forwarding, priority=0 actions=set_field:0x20/0xf0->reg0,goto_table:L2ForwardingCalc
 ```
 
-The following two flows send the traffic not matched by other flows to
-[SNATTable]. One of the flows is for egress traffic from local Pods; another
-one is for egress traffic from remote Pods, which is tunnelled to this Node to
-be SNAT'd with a SNAT IP configured on the Node. In the latter case, the flow
-also rewrites the destination MAC to the local gateway interface MAC.
+Flow 1 matches packets destined for the local Antrea gateway IP, rewrites their destination MAC address to that of the
+local Antrea gateway, loads `ToGatewayRegMark`, and forwards them to table [L3DecTTL] to decrease TTL value. It's worth
+noting that the action of rewriting the destination MAC address is not necessary but not harmful for Pod-to-gateway request
+packets because the destination MAC address is already the local gateway MAC address. However, it is utilized by the
+feature AntreaIPAM, which is not enabled by default.
```

Comment:
I meant that flow 1 is for AntreaIPAM, but we always enable it even when AntreaIPAM is not enabled.

Reply:
This is the meaning I wanted to express:

```
Flow 1 matches packets destined for the local Antrea gateway IP, rewrites their destination MAC address to that of the
local Antrea gateway, loads `ToGatewayRegMark`, and forwards them to table [L3DecTTL] to decrease TTL value. It's worth
noting that the action of rewriting the destination MAC address is not necessary but not harmful for Pod-to-gateway
request packets because the destination MAC address is already the local gateway MAC address. In short, the action is
only necessary for `AntreaIPAM` Pods, not required by the sample NodeIPAM Pods in this document.
```